### PR TITLE
[#3086] Remove support for deprecated configuration properties

### LINKED ISSUE #3086: Remove deprecated configuration properties
Several configuration properties have been marked as deprecated in the admin guides. These should be removed.

### PR DESCRIPTION
**Pull request by sophokles73**
Deprecated properties have been removed form configuration classes and
documentation.

Fixes #3086

### CODE REVIEW

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttProtocolAdapterProperties.java [hunk 1]
```diff
@@ -47,48 +47,9 @@ public MqttProtocolAdapterProperties() {
...
-     * <p>
-     * The default value is {@link #DEFAULT_COMMAND_ACK_TIMEOUT}.
-     *
-     * @deprecated Use {@link #getSendMessageToDeviceTimeout()} instead.
-     *
-     * @return The timeout in milliseconds.
-     */
-    @Deprecated(forRemoval = true)
-    public final int getCommandAckTimeout() {
-        return commandAckTimeout;
```
Comment (calohmn) [inline adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttProtocolAdapterProperties.java:68]: The `commandAckTimeout` field, along with `DEFAULT_COMMAND_ACK_TIMEOUT` and `getEffectiveSendMessageToDeviceTimeout()` should be removed as well.

Comment (sophokles73) [reply adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttProtocolAdapterProperties.java:68]: That seemed to have slipped through ... thanks for spotting :+1:

**Comment by sophokles73** [comment]
@calohmn I have pushed changes

**Review by calohmn — APPROVED** [review]
LGTM

**Pull request merged**